tools/@aws-cdk/remodel/lib/cli.ts

```diff
@@ -77,45 +83,199 @@ async function copyTemplateFiles(src: string, target: string) {
+
+    // May already exist as a result of codegen
+    await createIfNoDir(libDir);
+
+    // Only copy stable modules source code, L1s are built during codegen
+    if (library.packageJson.stability !== 'experimental') {
+      await transformPackage(library, newPkgJson, libDir, packagesToBundle, ubgConfig.monoPackageRoot);
+    }
+
+    if (library.shortName === 'core') {
+      indexStatements.push(`export * from './${library.shortName}';`);
+      lazyExports.unshift(`export * from './${library.shortName}';`);
+    } else {
+      const exportName = library.shortName.replace(/-/g, '_');
+
+      indexStatements.push(`export * as ${exportName} from './${library.shortName}';`);
+      lazyExports.push(`Object.defineProperty(exports, '${exportName}', { get: function () { return require('./${library.shortName}'); } });`);
+    }
+    copySubmoduleExports(pkgJsonExports, library, library.shortName);
+  }
+
+  // make the exports.ts file pass linting
+  lazyExports.unshift('/* eslint-disable @typescript-eslint/no-require-imports */');
+
+  await fs.writeFile(path.join(libRoot, 'index.ts'), indexStatements.join('\n'), { encoding: 'utf8' });
+  await fs.writeFile(path.join(libRoot, 'lazy-index.ts'), lazyExports.join('\n'), { encoding: 'utf8' });
 
-    return {
-      ...accum,
-      [p]: v,
-    };
-  }, {});
+  // Filter out packages we are bundling from dev dependencies
+  const packagesToBundleName = packagesToBundle.map(p => p.packageJson.name);
+  const deprecatedPackagesName = deprecatedPackages.map(p => p.packageJson.name);
+
+  // Filter out all of the stuff we don't want in devDeps anymore
+  const filteredDevDepsEntries = Object.entries(devDependencies)
+    .filter(
+      ([p]) => !(
+        packagesToBundleName.includes(p)
+        || deprecatedPackagesName.includes(p)
+        || p === '@aws-cdk/ubergen'
+      )
+    );
+  const filteredDevDeps = Object.fromEntries(filteredDevDepsEntries);
+
+  const newPkgJsonExports = formatPkgJsonExports(pkgJsonExports);
 
   await fs.writeFile(pkgJsonPath, JSON.stringify({
     ...pkgJson,
+    exports: {
+      ...newPkgJsonExports,
+    },
+    ubergen: {
+      ...pkgJson.ubergen,
+      libRoot,
+    },
     scripts: {
       ...pkgJson.scripts,
       'gen': 'ts-node scripts/gen.ts',
-      'build': 'yarn gen',
+      'build': 'yarn gen && cdk-build',
     },
     devDependencies: {
       ...filteredDevDeps,
       '@aws-cdk/cfn2ts': '0.0.0',
     },
   }, null, 2));
+
+  const rootFiles = await fs.readdir(awsCdkLibDir);
+  await Promise.all(rootFiles.map((file: string) => {
+    if (file.endsWith('.ts')) {
+      return fs.unlink(path.join(awsCdkLibDir, file));
+    }
+
+    return Promise.resolve();
+  }));
+
+  // await cleanupPackages(packagesToBundle);
+  // await cleanupPackages(deprecatedPackages);
+}
+
+// async function cleanupPackages(libraries: readonly LibraryReference[]) {
+//   for (const library of libraries) {
+//     await fs.remove(library.root);
+//   }
+// }
```

Comment:
Comment cleanup needed?

Reply:
yeah still WIP here. Will pull out of draft when ready for review.